scripts/swupd-completion.sh

```diff
@@ -0,0 +1,139 @@
+#!/bin/bash
+
+# This script creates a _swupd function to be used by
+# completion built-in bash command to complete swup
+# It reads options and subcommands from swup --help
+# Also it reads options for each subcommand recursively
+
+
+SCRIPTNAME="swupd.bash"
+COMPLETIONFUNCTION="_swupd"
+SWUPDCOMMAND="./swupd"
+
+optionsfromhelp=""
+subcommandsfromhelp=()
+optionsfromsubcommand=()
+
+#Checks if swupd commad exists in current shell
+if [ -a $SWUPDCOMMAND ] ; then
+    currenttype=""
+    nexttype=""
+    array=()
+    #reads "swupd --help" line by line
+    while read line
+    do
+	#For each line checks in what section is
+	if [[ $line == "Help Options:" ]]; then
+	    nexttype="options"
+	fi
+	if [[ $line == "Subcommands:" ]]; then
+	    nexttype="subcommand"
+	fi
+	if [[ $line == "" ]]; then
+	    currenttype=""
+	fi
+	case $currenttype in
```

Comment:
Since the value of $currenttype may be the empty string "", I would add another case for the empty string (even if you just run "break").